crates/networking/p2p/src/network/lean.rs

```diff
@@ -128,36 +137,189 @@ impl LeanNetworkService {
+            }
+            Err(err) => {
+                error!("Failed to start libp2p peer listen on {multi_addr:?}, error: {err:?}",);
+            }
+        }
+        self.connect_to_peers(peer_config.to_multiaddrs_lean())
+            .await;
         loop {
             tokio::select! {
                 Some(event) = self.swarm.next() => {
-                    if let Some(event) = self.parse_swarm_event(event).await {
+                    if let Some(event) = self.parse_lean_swarm_event(event).await {
                         info!("Swarm event: {event:?}");
                     }
                 }
             }
         }
     }
 
-    async fn parse_swarm_event(
+    async fn parse_lean_swarm_event(
         &mut self,
         event: SwarmEvent<ReamBehaviourEvent>,
     ) -> Option<ReamNetworkEvent> {
         match event {
-            SwarmEvent::Behaviour(_) => match None {
-                Some(ReamBehaviourEvent::Identify(_)) => None,
-                _ => None,
-            },
+            SwarmEvent::Behaviour(_) => None,
+            SwarmEvent::ConnectionEstablished { peer_id, .. } => {
+                self.peer_table
+                    .write()
+                    .insert(peer_id, ConnectionState::Connected);
+                info!("Connected to peer: {peer_id:?}");
+                None
+            }
+            SwarmEvent::ConnectionClosed { peer_id, .. } => {
+                self.peer_table
+                    .write()
+                    .insert(peer_id, ConnectionState::Disconnected);
+                info!("Disconnected from peer: {peer_id:?}");
+                Some(ReamNetworkEvent::PeerDisconnected(peer_id))
+            }
+            SwarmEvent::IncomingConnection {
+                local_addr,
+                send_back_addr,
+                ..
+            } => {
+                info!("Incoming connection from {send_back_addr:?} to {local_addr:?}");
+                None
+            }
+            SwarmEvent::Dialing { peer_id, .. } => {
+                info!("Dialing {peer_id:?}");
+                Some(ReamNetworkEvent::PeerConnectedOutgoing(peer_id?))
+            }
+            SwarmEvent::OutgoingConnectionError { peer_id, error, .. } => {
+                warn!("Failed to connect to {peer_id:?}: {error:?}");
+                None
+            }
             _ => None,
         }
     }
+
+    async fn connect_to_peers(&mut self, peers: Vec<Multiaddr>) {
+        trace!("Discovered peers: {peers:?}");
+        for peer in peers {
+            let mut successfully_dialed = false;
+            if let Err(err) = self.swarm.dial(peer.clone()) {
+                warn!("Failed to dial peer: {err:?}");
+            } else {
+                successfully_dialed = true;
+            }
+
+            if !successfully_dialed {
+                trace!("Failed to dial any multiaddr for peer: {peer:?}");
+                continue;
+            }
```

Comment:
```suggestion
            if let Err(err) = self.swarm.dial(peer.clone()) {
                warn!("Failed to dial peer: {err:?}");
                continue;
            }
```

^ it looks like this was just copy pasted without reading the code, I cleaned it up